File: 2523_closest_prime_numbers_in_range.java
import java.util.*;

class Solution {
    public int[] closestPrimes(int a, int b) {
        boolean[] x = wobble(b + 1);
        List<Integer> y = new ArrayList<>();
        for (int i = a; i <= b; i++) if (x[i]) y.add(i);
        if (y.size() < 2) return new int[]{-1, -1};
        int minDiff = Integer.MAX_VALUE, u = -1, v = -1;
        for (int i = 1; i < y.size(); i++) {
            int d = y.get(i) - y.get(i - 1);
            if (d < minDiff) { minDiff = d; u = y.get(i - 1); v = y.get(i); }
        }
        return new int[]{u, v};
    }

    private boolean[] wobble(int n) {
        boolean[] z = new boolean[n];
        Arrays.fill(z, true);
        z[0] = z[1] = false;
        for (int i = 2; i * i < n; i++) if (z[i]) 
            for (int j = i * i; j < n; j += i) z[j] = false;
        return z;
    }
}

File: 2523_closest_prime_numbers_in_range.py
import math

class Solution:
    def closestPrimes(self, a, b):
        x = self.wobble(b + 1)
        y = [i for i in range(a, b + 1) if x[i]]
        return min([(v - u, [u, v]) for u, v in zip(y, y[1:])], default=[[math.inf, [-1, -1]]])[1] if len(y) > 1 else [-1, -1]
    
    def wobble(self, n):
        z = [1] * n
        z[:2] = [0, 0]
        for i in range(2, int(n**0.5) + 1):
            if z[i]: z[i*i:n:i] = [0] * len(z[i*i:n:i])
        return z
